File: breaking-records/breaking-records.js
// https://www.hackerrank.com/challenges/breaking-best-and-worst-records/problem

const breakingRecords = scores => {
  const records = scores.reduce(
    ({ high, low, newHighs, newLows }, score) => ({
      high: score > high ? score : high,
      low: score < low ? score : low,
      newHighs: score > high ? ++newHighs : newHighs,
      newLows: score < low ? ++newLows : newLows
    }),
    { high: scores[0], low: scores[0], newHighs: 0, newLows: 0 }
  );

  const { newHighs, newLows } = records;

  return [newHighs, newLows];
};

File: breaking-records/breaking-records.clj
(defn breakingRecords [scores]
  (let [records-start
        {:high (first scores)  :low (first scores) :new-highs 0 :new-lows 0}
        records (reduce
                 (fn
                   [{h :high, l :low, new-hs :new-highs, new-ls :new-lows} s]
                   {:high (if (> s h) s h)
                    :low (if (< s l) s l)
                    :new-highs (if (> s h) (inc new-hs) new-hs)
                    :new-lows (if (< s l) (inc new-ls) new-ls)})
                 records-start
                 scores)]
    [(get records :new-highs) (get records :new-lows)]))
